x-pack/plugins/ml/server/lib/ml_client/ml_audit_logger.ts

```diff
@@ -0,0 +1,383 @@
+        const [jobId] = getADJobIdsFromRequest(p);
+        return { message: `Closing anomaly detection job ${jobId}`, type: [EVENT_TYPES.change] };
+      }
+      case 'ml_update_ad_job': {
+        const [jobId] = getADJobIdsFromRequest(p);
+        return { message: `Updating anomaly detection job ${jobId}`, type: [EVENT_TYPES.change] };
+      }
+      case 'ml_reset_ad_job': {
+        const [jobId] = getADJobIdsFromRequest(p);
+        return { message: `Resetting anomaly detection job ${jobId}`, type: [EVENT_TYPES.change] };
+      }
+      case 'ml_revert_ad_snapshot': {
+        const [jobId] = getADJobIdsFromRequest(p);
+        const [params] = p as Parameters<MlClient['revertModelSnapshot']>;
+        const snapshotId = params.snapshot_id;
+        return {
+          message: `Reverting anomaly detection snapshot ${snapshotId} in job ${jobId}`,
+          type: [EVENT_TYPES.change],
+        };
+      }
+      case 'ml_put_ad_datafeed': {
+        const [datafeedId] = getDatafeedIdsFromRequest(p);
+        const [jobId] = getADJobIdsFromRequest(p);
+        return {
+          message: `Creating anomaly detection datafeed ${datafeedId} for job ${jobId}`,
+          type: [EVENT_TYPES.creation],
+        };
+      }
+      case 'ml_delete_ad_datafeed': {
+        const [datafeedId] = getDatafeedIdsFromRequest(p);
+        return {
+          message: `Deleting anomaly detection datafeed ${datafeedId}`,
+          type: [EVENT_TYPES.deletion],
+        };
+      }
+      case 'ml_start_ad_datafeed': {
+        const [datafeedId] = getDatafeedIdsFromRequest(p);
+        return {
+          message: `Starting anomaly detection datafeed ${datafeedId}`,
+          type: [EVENT_TYPES.change],
+        };
+      }
+      case 'ml_stop_ad_datafeed': {
+        const [datafeedId] = getDatafeedIdsFromRequest(p);
+        return {
+          message: `Stopping anomaly detection datafeed ${datafeedId}`,
+          type: [EVENT_TYPES.change],
+        };
+      }
+      case 'ml_update_ad_datafeed': {
+        const [datafeedId] = getDatafeedIdsFromRequest(p);
+        return {
+          message: `Updating anomaly detection datafeed ${datafeedId}`,
+          type: [EVENT_TYPES.change],
+        };
+      }
+      case 'ml_put_calendar': {
+        const [params] = p as Parameters<MlClient['putCalendar']>;
+        const calendarId = params.calendar_id;
+        // @ts-expect-error body is optional
+        const jobIds = (params.body ?? params).job_ids;
+        return {
+          message: `Creating calendar ${calendarId} ${jobIds ? `with job(s) ${jobIds}` : ''}`,
+          type: [EVENT_TYPES.creation],
+        };
+      }
+      case 'ml_delete_calendar': {
+        const [params] = p as Parameters<MlClient['deleteCalendar']>;
+        const calendarId = params.calendar_id;
+        return { message: `Deleting calendar ${calendarId}`, type: [EVENT_TYPES.deletion] };
+      }
+      case 'ml_put_calendar_job': {
```

Comment:
Is this creating the calendar? Just checking if `creation` is correct for the type here, or if it should be `change`?

Reply:
This is assigning a job to a calendar. I think it should be `change`

Reply:
Yes - `ml_put_calendar_job` and `ml_delete_calendar_job` modify the calendar, so they should be `change`.
`ml_post_calendar_events` - this is adding a new event with the reference to a calenadar(s), so that should be `create` and `ml_delete_calendar_event` should be `delete`.

Reply:
Updating in https://github.com/elastic/kibana/pull/195120/commits/26b9675ba9a2435db81a005de08b26b90fc0893d and https://github.com/elastic/kibana/pull/195120/commits/15a474634f26a7162809b99a15cbab70ab1d892b